### PR TITLE
+ PatientSignUpView

### PR DESCRIPTION
**Pull request by kamilkazmierczak**
Tam masz kilka komentarzy z pytaniami, nie robiłem rejestracji dla lekarza bo mi się nie chciało - zrobię to jak będę miał chwile czasu.
----
Co istotne to jest zrobione na starej wersji z błędną stroną startową bo na twoim gicie jej nie było

### CODE REVIEW

**Comment by tomasztomys** [comment]
Ciekawe podejście :)

File: front/src/views/PatientSignUpView/patient_sign_up_view.scss [hunk 1]
```diff
@@ -0,0 +1,49 @@
+.root{
+
+  .title-container{
+    position: relative;
+    text-align: center;
+    margin: 0 auto;
+
+      .registration-icon{
+        position: relative;
+      	width: 3em;
```
Comment (tomasztomys) [inline front/src/views/PatientSignUpView/patient_sign_up_view.scss]: wrong indent

**Comment by tomasztomys** [comment]
Dodałeś chyba przez przypadek plik .zip ;)

File: front/src/views/PatientSignUpView/patient_sign_up_view.scss (hunk 1 above)
Comment (kamilkazmierczak) [inline front/src/views/PatientSignUpView/patient_sign_up_view.scss]: nie ma linta czy jak to się tam nazywa do css więc robie jak mi się podoba

**Comment by kamilkazmierczak** [comment]
trudno się mówi

File: front/src/views/PatientSignUpView/PatientSignUpView.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          <Input text="Username" />
+          <Input text="First Name" />
+          <Input text="Last Name" />
+          <Input text="E-mail Address" />
+          <Input text="Password" />
+          <Input text="Repeat password" />
+
+          <div style={ buttonStyle }>
+              {
+                // dlaczego jak dodam te style do Button
```
Comment (dariuszpaluch) [inline front/src/views/PatientSignUpView/PatientSignUpView.js]: It's very poor idea to write polish comments in js file.

File: front/src/views/PatientSignUpView/patient_sign_up_view.scss [hunk 3]
```diff
@@ -0,0 +1,49 @@
+.root{
+
+  .title-container{
```
Comment (dariuszpaluch) [inline front/src/views/PatientSignUpView/patient_sign_up_view.scss]: add name before '{'

File: front/src/views/PatientSignUpView/PatientSignUpView.js (hunk 2 above)
Comment (dariuszpaluch) [inline front/src/views/PatientSignUpView/PatientSignUpView.js]: Better Idea is comments your code in PR, no in files

File: front/src/views/PatientSignUpView/PatientSignUpView.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+          </h1>
+        </div>
+
+        <div className={ style['user-input-container'] }>
+
+          <Input text="Username" />
+          <Input text="First Name" />
+          <Input text="Last Name" />
+          <Input text="E-mail Address" />
+          <Input text="Password" />
```
Comment (nkaczor) [inline front/src/views/PatientSignUpView/PatientSignUpView.js]: You have special component PasswordInput. Please don't write two component in one file.  If you want have this in one file I recommend you to use new render function such as renderInputWithLabel(args)
Although naming this component Input doesn't appeal to me. We have something similar to it. TextInput!

File: front/src/views/PatientSignUpView/PatientSignUpView.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
+import React, { Component } from 'react';
+
+import style from './patient_sign_up_view.scss';
+import { Button, TextInput } from '../../components';
+import Svg from 'svg-inline-react';
+import registrationIcon from '../../../assets/icons/user-tie.svg';
+
```
Comment (nkaczor) [inline front/src/views/PatientSignUpView/PatientSignUpView.js]: Please don't write two component in one file.  If you want have this in one file I recommend you to use new render function such as renderInputWithLabel(args)
Although naming this component Input doesn't appeal to me. We have something similar to it. TextInput!

**Comment by nkaczor** [comment]
Please add password confirmation :)

File: front/src/views/PatientSignUpView/PatientSignUpView.js [hunk 6]
```diff
@@ -1,32 +1,31 @@
...
-            { this.props.text }:
+            { label }:
         </h3>
         <div className={ style['input-container'] }>
           <div className="row">
-            {
-              // jbc to ten row + col-xs mi sie tu nie podoba trzeba to zmienic
-              // ale inaczej wylazi poza input-container plus jest na 100% szerokosci
+          { (() => {
+            switch(inputType) {
```
Comment (kamilkazmierczak) [inline front/src/views/PatientSignUpView/PatientSignUpView.js]: I can put here simple if statement if you don't like this, but it works for me.

Comment (tomasztomys) [inline front/src/views/PatientSignUpView/PatientSignUpView.js]: Omg,  what is it?! Add a new component, for example use React.Children

File: front/src/views/PatientSignUpView/PatientSignUpView.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+  renderInputWithLabel(inputType, label) {
+    return (
+      <div className={ style['simple-input-box'] }>
+        <h3 className={ style['input-label'] }>
+            { label }:
+        </h3>
+        <div className={ style['input-container'] }>
+          <div className="row">
+          { (() => {
```
Comment (tomasztomys) [inline front/src/views/PatientSignUpView/PatientSignUpView.js]: What is it?!!! Add new component,  for example use React.Children

Comment (nkaczor) [inline front/src/views/PatientSignUpView/PatientSignUpView.js]: No, we have components: TextInput and Password Input. I would leave it as it is, but this part with lambda function and switch is a kind of scary :D. So please for example do something like this:
`let input = {
text:<TextInput className="col-xs-12" />,
password: <PasswordInput className="col-xs-12" /> 
}`

and then you can use it this way in JSX: 
`{input[inputType]}`

**Comment by nkaczor** [comment]
Why are you using labels? We don't use it in SignInView and I think we should be consistent.
And why are you using rows and columns, if you set them to 100% width?

**Pull request merged**